### PR TITLE
Code cleanups

### PR DESCRIPTION
**Pull request by dahliamalkhi**
This branch contains stylistic code cleanups. 

no functionality change -intended-.

### CODE REVIEW

**Comment by Maithem** [comment]
@dahliamalkhi  I think the commit history can be improved. Please squash commits, remove your local merge commit and add git commit messages.

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiSMREntry.java [hunk 1]
```diff
@@ -64,6 +69,8 @@ public void setEntry(ILogData entry) {
 
     @Override
     public List<SMREntry> getSMRUpdates(UUID id) {
+        // FIXME MultiSMREntry should not implement ISMRConsumable; we have MultiObjectSMREntry for that
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiSMREntry.java]: Any consumable SMR entry should implement ISMRConsumable, I don't see how MultiSMREntry should be any different!

File: runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java [hunk 2]
```diff
@@ -66,7 +66,7 @@
      * A view of streams in the Corfu server instance.
      */
     @Getter(lazy = true)
-    private final MultiStreamView streamsView = new MultiStreamView(this);
+    private final MultiStreamView multiStreamsView = new MultiStreamView(this);
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java]: this is getting out of hand. I have an idea, we can keep the classes named the name, but the fields shorter, sane names:

runtime.layout()
runtime.streams()
runtime.objects()
runtime.addressSpace()

-etc-

File: runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java [hunk 3]
```diff
@@ -79,6 +79,7 @@
     public final TransactionBuilder builder;
 
     /**
+     * TODO remove this!
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java]: Why remove it? It seems like it may be useful to keep track of the running time of a transaction - and either way shouldn't have much overhead.

File: runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java (hunk 2 above)
Comment (dahliamalkhi) [inline runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java]: agreed, though 'layout' is already taken.

Comment (dahliamalkhi) [inline runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java]: for now, though, let's keep all names. don't want to break applications.

File: runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java (hunk 3 above)
Comment (dahliamalkhi) [inline runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java]: we have metering for that. this is completely isolated mechanism, mostly unknown and hiding. No reason to have it at this level.

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiSMREntry.java (hunk 1 above)
Comment (dahliamalkhi) [inline runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiSMREntry.java]: ok.

**Comment by dahliamalkhi** [comment]
@Maithem  done.

File: runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java (hunk 3 above)
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java]: Lots of reasons to have it at this level, for example, metering won't give you per transaction level information, which you may very well want to introspect on (like self aborting if the transaction is too old...)

File: runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java (hunk 2 above)
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java]: Ok, then you're going to undo this change, right? If you're going to break applications, you might as well produce the intended change...

File: runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java [hunk 4]
```diff
@@ -66,7 +66,7 @@
      * A view of streams in the Corfu server instance.
      */
     @Getter(lazy = true)
-    private final MultiStreamView multiStreamsView = new MultiStreamView(this);
+    private final StreamsView streams = new StreamsView(this);
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java]: I think if you're going to do this, you should get rid of the get prefix on the autogenerated getter (no need to make it lazy, that was a remenant if the past...)

File: runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java (hunk 3 above)
Comment (slfritchie) [inline runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java]: The completion time is currently used for a log.trace() message, I think.  If making it more official'y via a new metering statistic is useful, I'm fine with that.  I'll let y'all armwrestle about wisdom of other uses.

Comment (dahliamalkhi) [inline runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java]: is there anything that can't be done at application level (before TXbegin and post TXend)?

File: runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java [hunk 5]
```diff
@@ -344,3 +274,74 @@ public String toString() {
...
+
+    public Set<Integer> getConflictSet(UUID streamID) {
+        return getReadSetConflicts().computeIfAbsent(streamID, u -> {
+            return new HashSet<>();
+        } );
+    }
+
+}
+
+@Getter @groovy.util.logging.Slf4j
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java]: This appears to be the wrong slf4j annotation (you want Lombok not groovy)

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java [hunk 6]
```diff
@@ -17,27 +17,37 @@
  * Created by mwei on 9/20/16.
  */
 @ToString
-@NoArgsConstructor
 @Slf4j
-public class MultiObjectSMREntry extends LogEntry implements IDivisibleEntry, ISMRConsumable {
+public class MultiObjectSMREntry extends LogEntry implements ISMRConsumable {
```
Comment (dahliamalkhi) [inline runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java]: ok, I get penalized for my good deeds...

 I'll add some comments.
